src/scipp/testing/assertions.py

```diff
@@ -31,6 +32,25 @@
 T = TypeVar('T')
 
 
+def assert_allclose(a: T, b: T, **kwargs) -> None:
```

Comment:
The `kwargs` are numpy arguments, right? In `sc.allclose` we require the tolerances to be `Variables`. So we should do the same here.

Reply:
This is a bit tricky, we can require them to be `Variable` but should we respect units?
For example if I compare two `DataArrays` the coords and the data have different units, so the same `atol` variable cannot be used for both.
How is that handled in `sc.allclose`, or does that only compare `Variable`?

One option is to require them to be `Variables` but then disregard the units.
For the `rtol` it probably doesn't matter since it's always dimensionless anyway.

Reply:
The coords have to match exactly. The approximate comparison is only applied to the data. So the unit of `atol` must be the same as the data and `rtol` must be dimensionless.

Reply:
Okay that makes sense.

But there is a similar issue for datasets and datagroups. In that case we have multiple `data` fields (say the DataGroup contains 2 DataArray objects for example) and they might have different units.

Reply:
Those arrays cannot be close. So just signal a unit mismatch error. This should happen before the values are even compared.

Reply:
Hm I don't understand. Should the `assert_allclose` not be applicable to datasets and datagroups?